hudi-utilities/src/test/java/org/apache/hudi/utilities/schema/SchemaRegistryProviderTest.java

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utilities.schema;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.avro.Schema;
+import org.apache.hudi.common.config.TypedProperties;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+class SchemaRegistryProviderTest {
+
+  private final String basicAuth = "foo:bar";
+
+  private final String json = "{\"schema\":\"{\\\"type\\\": \\\"record\\\", \\\"namespace\\\": \\\"example\\\", "
+      + "\\\"name\\\": \\\"FullName\\\",\\\"fields\\\": [{ \\\"name\\\": \\\"first\\\", \\\"type\\\": "
+      + "\\\"string\\\" }]}\"}";
+
+  private TypedProperties getProps() {
+    return new TypedProperties() {{
+        put("hoodie.deltastreamer.schemaprovider.registry.baseUrl", "http://" + basicAuth + "@localhost");
+        put("hoodie.deltastreamer.schemaprovider.registry.urlSuffix", "-value");
+        put("hoodie.deltastreamer.schemaprovider.registry.url", "http://foo:bar@localhost");
+        put("hoodie.deltastreamer.source.kafka.topic", "foo");
+      }};
+  }
+
+  Schema getExpectedSchema(String response) throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    JsonNode node = mapper.readTree(new ByteArrayInputStream(response.getBytes(StandardCharsets.UTF_8)));
+    return (new Schema.Parser()).parse(node.get("schema").asText());
+  }
+
+  @Test
+  public void testGetSourceSchemaShouldRequestSchemaWithCreds() throws IOException {
+    InputStream is = new ByteArrayInputStream(json.getBytes(StandardCharsets.UTF_8));
+    SchemaRegistryProvider underTest = new SchemaRegistryProvider(getProps(), null);
+    SchemaRegistryProvider spyUnderTest = Mockito.spy(underTest);
+    Mockito.doReturn(is).when(spyUnderTest).getStream(Mockito.any());
+    Schema actual = spyUnderTest.getSourceSchema();
+    assertNotNull(actual);
+    assertEquals(actual, getExpectedSchema(json));
+    verify(spyUnderTest, times(1)).setAuthorizationHeader(eq(basicAuth),
+        Mockito.any(HttpURLConnection.class));
+  }
+
+  @Test
+  public void testGetTargetSchemaShouldRequestSchemaWithCreds() throws IOException {
+    InputStream is = new ByteArrayInputStream(json.getBytes(StandardCharsets.UTF_8));
```

Comment:
Can we re-use code across previous test and this one. except for 1 or 2 lines, mostly its same. we can create a private method w/ an argument for whether its source schema or target schema

Reply:
done